rfcs/20210219-cache-compiled-gpu-kernels/README.md

```diff
@@ -0,0 +1,101 @@
+# Proposal on caching GPU kernels instead of their binary representations
+
+Motivation
+
+The primitive cache is designed to hold the whole primitive which can contain
+CPU or GPU kernel. The GPU kernel can have two states:
+1. Compiled kernel
+2. Binary representation of the kernel
+
+Currently, GPU primitive that is stored in the primitive cache contains GPU kernel
+which has the 2nd state.
+
+There are two cons of such approach:
+1. Binary representation is basically an array of bytes hence no 3rd party resources
+are stored in the primitive cache
+2. Binary representation is associated with a particular GPU device but is context
+independent therefore it can be reused for different GPU contexts
```

Comment:
why is this a cons? It actually could be a pros to have a binary independent  (e.g. it could be reused for an OCL device or for a level_zero device).

Reply:
It's pros, sorry for the mistake.